File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int ans=nums[0];
        int total=0;
        for(int i=0;i<nums.length;i++){
            if(total<0){
                total=0;
            }
            total=total+nums[i];
            ans=Math.max(ans,total);

        }
        return ans;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sum=0
        ans = -sys.maxsize-1
        for i in nums:
            sum=sum+i
            ans=max(ans,sum)
            if(sum<0):
                sum=0
        return ans
        